### DISCUSSION TITLE
isn't plugin "ubuntu" poorly named?

### DISCUSSION THREAD

**Discussion by jjangga0214** [ohmyzsh/ohmyzsh#11972] [Ideas]
Currently, most features of plugin **ubuntu** are in fact not for ubuntu itself but for "apt".

Apt is not ubuntu-specific. It's used on debian-based distro, and even other distro not based on debian.  

In my opinion, name of plugin ought to imply what it actually does and serves.

So, i think, first, it should be renamed, and second, I'd like to suggest renaming it to **apt**. Though not 100% of this plugin is direct concern of apt (e.g. `kerndeb`, but even it is also indirectly related with apt), its almost entire purpose is for using apt with ease. So this rename might make sense. (Or, we can separate ubuntu-related things and apt-related things each for different plugins, and organize them in future.)

And yes, this change forces users to edit .zshrc. However, this inconvenience might be resolved in a soft and adaptive ways, such as notification after update or fall back to renamed plugin with warning(guide message to change) for a while.

**Comment by mcornella** [comment]
I don't think it's a bad name. The two apt-based main distributions are debian and ubuntu, and it makes sense keeping them separate for the time being. And as you said, `apt` is not the only program being used in the plugin, there is also `apt-get` and others, so `apt` is not really a suitable name.

**Comment by jjangga0214** [comment]
OK @mcornella, you definitely have a different opinion.
So, why don't we have a little discussion about several points?

First, I'd like to suggest a failing scenario with name "ubuntu". It's highly expected most of users won't read READMEs of all plugins. Rather, they would 1) look over the list of plugin names, 2) and then read README of ONLY interesting plugin. Users of debian-based, but not ubuntu-based disto like Kali Linux might think plugin "ubuntu" is not for them. Of course, they can read README so they understand the plugin could be useful on their os as well, but they simply wouldn't, just because of its name. What's more, even some users of ubuntu-based disto like Linux Mint also think the same, which results their not using the plugin. 

Then what would happen with name "apt"? While looking though the plugin list, users find name "apt", and they probably think they should read README to know if they can use it. As users read it, they will find the plugin useful. So, naming as "apt" is not an obstacle to user influx, unlike "ubuntu".

The most important virtue of plugin (or opensource software in general) is to be used as widely and properly as possible. In this perspective, "apt" is better than "ubuntu". 

Second, I hereby make sure that the plugin name "apt" actually doesn't mean command `apt`, but packaging tool 'apt'. They are different. Debian uses 'apt' (advanced packaging tools) to manage its packaging system. There are various tools that interact with 'apt' and allow users to install, remove and manage packages in debian-based distro. `apt-get` is one such command tool popular. Another popular tool is `aptitude` with both GUI and command-line options. So are `apt-cache` and etc. As the most commonly used package management commands are scattered (especially across `apt-get` and `apt-cache`), command `apt` is introduced to provide consistent command interface. 

> there is also apt-get and others, so apt is not really a suitable name.

So, we don't have to worry if plugin name "apt" is suitable for command `apt-get`. Regardless of differences between commands used in the plugin such as `apt`, `apt-get`, `apt-cache`, `apt-file`, `apt-add-repository`, `ppa-purge`, `aptitude`, `dpkg`, `dpkg-buildpackage`, and `dpkg-query`, they are all for managing 'apt' with ease, so plugin name "apt" would make sense. 

One thing I'm not sure of is whether `kerndeb` interacts with 'apt' (but at least it would be somehow indirectly related with 'apt'?).

**Comment by jjangga0214** [comment]
@mcornella ping?

**Comment by mcornella** [comment]
Reopening for the moment, will discuss later.

**Comment by mcornella** [comment]
Resurrecting this. This is a big change, so it needs big user feedback.

@jkemp814 [suggests](https://github.com/ohmyzsh/ohmyzsh/issues/10359) also merging `debian` and `ubuntu` plugins into a single `apt` plugin.

**Comment by jkemp814** [comment]
I had not found this proposal before posting about this issue. It seems right on the mark. The two plugins are redundant. They both do the exact same thing, i.e. aliases for apt. (with Ubuntu plugin having the bulk of the aliases) 

Consolidating the two plugins would tidy up the plugins list, and serve a broader Debian based and Ubuntu derivatives Linux community. There are a whole list of Distros that could use a plugin named 'apt.' Right now, those users would have to choose to use a plugin designated for another distro. 

Those aliases have nothing to do with Debian or Ubuntu, with the exception they both use apt for package management. APT is really apt-get. Aptitude is its own package but does the exact same thing in most of its functions. The naming of the plugin apt is short and right to the point of the two plugins purpose. Having a few other aliases in there about package management would not hurt anything. The way it is now is way off the mark, and neither does anything as far as the OS itself.

Think of it like the 'brew' plugin has its own. Maybe we can move all the aliases pertaining too 'apt' to a new plugin and see what's left in the Debian and Ubuntu plugins.
